src/library_manager/lib_manager.c

```diff
@@ -445,19 +445,28 @@ static int lib_manager_dma_deinit(struct lib_manager_dma_ext *dma_ext, uint32_t
 static int lib_manager_load_data_from_host(struct lib_manager_dma_ext *dma_ext, uint32_t size)
 {
 	struct dma_status stat;
+	int loop_count = 1000;
 	int ret;
 
 	ret = dma_get_status(dma_ext->chan->dma->z_dev, dma_ext->chan->index, &stat);
 	if (ret < 0)
 		return ret;
 
 	/* Wait till whole data acquired */
-	while (stat.pending_length < size) {
+	while (stat.pending_length < size && loop_count) {
```

Comment:
better to adjust the sleep time instead of add new count, if add new count, how to determine the times?
100 us * 1000 = 100ms, so you are expect load library within 100ms?

Reply:
I thought that it is a bit of over-engineering to have time check, so I used a simple counter.
Let me see if I can find a good template for the time counter.